src/Entities/DreamStuff/DreamFallingBlock.cs

```diff
@@ -15,11 +35,39 @@ public class DreamFallingBlock : CustomDreamBlock {
         public bool HasStartedFalling { get; private set; }
         private bool hasLanded;
 
+        private DreamFallingBlockChainRenderer chainRenderer;
+        private bool chained;
+        private bool held;
+
+        private float chainStopY, startY;
+        private bool centeredChain;
+        private bool chainOutline;
+
+        private bool indicator, indicatorAtStart;
+        private float pathLerp;
+
+        private SoundSource rattle;
+
         public DreamFallingBlock(EntityData data, Vector2 offset)
             : base(data.Position + offset, data.Width, data.Height, data.Bool("featherMode"), data.Bool("oneUse"), GetRefillCount(data), data.Bool("below")) {
             noCollide = data.Bool("noCollide", false);
 
+            chained = data.Bool("chained");
```

Comment:
would it be feasible to implement this as a subclass of `DreamFallingBlock` instead?

Reply:
I'd like to do this for 1.9.1, we don't have time anymore

Reply:
This will be worked on for #69.